src/seedu/addressbook/commands/Command.java

```diff
@@ -70,4 +70,8 @@ public int getTargetIndex() {
     public void setTargetIndex(int targetIndex) {
         this.targetIndex = targetIndex;
     }
+
+    public boolean isMutating() {
+        throw new UnsupportedOperationException("This method should be implement in child classes");
```

Comment:
there is a cleaner way of doing this. think about what kind of class `Command` is.